include/drivers/kscan.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for Keyboard scan matrix devices.
+ * The scope of this API is simply to report which key event was triggered
+ * and users can later decode keys using their desired scan code tables in
+ * their application. In additon, typematic rate and delay can easily be
+ * implemented using a timer if desired.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_KB_SCAN_H_
+#define ZEPHYR_INCLUDE_DRIVERS_KB_SCAN_H_
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief KSCAN APIs
+ * @defgroup kscan_interface Keyboard Scan Driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @brief Keyboard scan callback called when user press/release
+ * a key on a matrix keyboard.
+ *
+ * @param dev   Pointer to the device structure for the driver instance.
+ * @param data  Data byte passed pack to the user.
+ * @param col	Describes column change.
+ * @param row	Describes row change.
+ * @param pressed Describes the kind of key event.
+ */
+typedef void (*kscan_callback_t)(struct device *dev, u8_t row, u8_t column,
+				 bool pressed);
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * Keyboard scan driver API definition and system call entry points.
+ *
+ * (Internal use only.)
+ */
+typedef int (*kscan_config_t)(struct device *dev,
+			    kscan_callback_t callback_isr);
```

Comment:
consider renaming callback_isr to callback since this is not really an isr

Reply:
agreed, this runs in a worker thread context

Reply:
renamed to callback